src/foam/lib/csv/CSVParser2.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.lib.csv',
+  name: 'CSVParser2',
+  requires: [
+    'foam.parse.StringPStream'
+  ],
+  grammars: [
+    {
+      name: 'csvRow',
+      symbols: function() {
+        return {
+          START: seq1(0, repeat(sym('field'), ',')),
+          field: alt(sym('quotedText'), sym('unquotedText'), ''),
+          unquotedText: repeat(not(alt(',','\n', '\r'), anyChar()), '', 1),
+          quotedText: seq1(1, '"', repeat(alt(sym('escapedQuote'), not('"', anyChar()))), '"'),
+          escapedQuote: '""'
+        };
+      },
+      actions: {
+        unquotedText: function (a) {
+          return a.join('');
+        },
+        quotedText: function (a) {
+          return a.join('');
+        },
+        escapedQuote: function () { return '"'; }
```

Comment:
Why is this an action instead of a symbol:?

Reply:
It's kind of subtle to see but this replaces `""` (i.e. two double quotes) with just one `"`.